packages/opensrp-plans/src/components/MissionData/index.tsx

```diff
@@ -0,0 +1,89 @@
+/** renders card that shows mission indicators info and mission data csv export */
+import {
+  FIX_PRODUCT_PROBLEMS_CODE,
+  PlanDefinition,
+  PRODUCT_CHECK_CODE,
+  SERVICE_POINT_CHECK_CODE,
+} from '@opensrp/plan-form-core';
+import React, { useEffect, useState } from 'react';
+import { Card, Button, Typography, Space, Alert } from 'antd';
+import {
+  DOWNLOAD_MISSION_DATA,
+  FETCHING_MISSION_INDICATORS_DATA,
+  MISSION_DATA,
+  NUMBER_OF_FLAGGED_PRODUCTS,
+  PRODUCTS_CHECKED,
+  SERVICE_POINTS_VISITED,
+} from '../../lang';
+import { loadTasksIndicators, TaskCount } from '../../helpers/dataLoaders';
+import { CommonProps, defaultCommonProps } from '@opensrp/plan-form';
+import { useHandleBrokenPage } from '@opensrp/react-utils';
+
+const { Title, Text } = Typography;
+
+interface MissionDataProps extends CommonProps {
+  plan: PlanDefinition;
+}
+
+const defaultProps = {
+  ...defaultCommonProps,
+};
+
+const MissionData = (props: MissionDataProps) => {
+  const { plan, baseURL } = props;
+  const [servicePoints, setServicePoints] = useState<string>('');
+  const [productsChecked, setProductsChecked] = useState<string>('');
+  const [flaggedProducts, setFlaggedProducts] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(true);
+  const { handleBrokenPage, broken, errorMessage } = useHandleBrokenPage();
+
+  useEffect(() => {
+    const { identifier: planId } = plan;
+    const promises: Promise<void | Response>[] = [];
+    const codes = [SERVICE_POINT_CHECK_CODE, PRODUCT_CHECK_CODE, FIX_PRODUCT_PROBLEMS_CODE];
+    const setStateSequence = [setServicePoints, setProductsChecked, setFlaggedProducts];
+    codes.forEach((code, index) => {
+      const thisPromise = loadTasksIndicators(baseURL, planId, code, true)
+        .then((response: TaskCount) => {
+          setStateSequence[index](response.total_records);
+        })
+        .catch((e) => e);
```

Comment:
throw an error, here, I noticed for some reason returning an error does not result in a proper Promise.reject